### PR TITLE
Feature: App-531 Gas fee estimation for deposit

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Add deposit gas fee estimation
- replace deposit modal dummy data with real data

************************************************************
IMPORTANT NOTE: 
the estimation for erc20s needs token allowance, this issue needs to be addressed by SDK.  
Fixing it on SDK is ongoing!
************************************************************

Task: [APP-531](https://aragonassociation.atlassian.net/browse/APP-531)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/containers/transactionModals/DepositModal.tsx [hunk 1]
```diff
@@ -44,9 +67,43 @@ const DepositModal: React.FC<TransactionModalProps> = ({
...
+        ) as string);
+
+  useEffect(() => {
+    setUpdateBlink(true);
+    setTimeout(() => {
+      setUpdateBlink(false);
+    }, 1000);
+  }, [averageFee]);
+
+  useEffect(() => {
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/transactionModals/DepositModal.tsx]: There is a `tokenPrice` already from `usePollGasFee`. Is there an issue with that one?

File: packages/web-app/src/containers/transactionModals/DepositModal.tsx [hunk 2]
```diff
@@ -55,10 +112,39 @@ const DepositModal: React.FC<TransactionModalProps> = ({
...
+    () =>
+      averageFee === undefined
+        ? ['Error calculating costs', 'Error estimating fees']
+        : [
+            new Intl.NumberFormat('en-US', {
+              style: 'currency',
+              currency: 'USD',
+            }).format(
+              Number(
+                formatUnits(averageFee.toString(), nativeCurrency.decimals)
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/transactionModals/DepositModal.tsx:134]: You forgot to add the deposit amount to the average to get the total cost.

**Review by Fabricevladimir — APPROVED** [review]
Thanks Sepehr. Left a couple of comments

**Review by RakeshUP — APPROVED** [review]

File: packages/web-app/src/containers/transactionModals/DepositModal.tsx (hunk 1 above)
Comment (sepehr2github) [reply packages/web-app/src/containers/transactionModals/DepositModal.tsx]: That's token price used for eth price (for gas fee), this one is for selected token price

**Pull request merged**